babel/src/main/java/org/apache/calcite/sql/babel/SqlExecMacro.java

```diff
@@ -32,31 +33,63 @@
 import java.util.Objects;
 
 /**
- * Parse tree for {@code EXEC} statement.
+ * Parse tree for {@code SqlExecMacro} statement.
  */
 public class SqlExecMacro extends SqlCall implements SqlExecutableStatement {
   public static final SqlSpecialOperator OPERATOR =
       new SqlSpecialOperator("EXECUTE", SqlKind.OTHER);
 
   private final SqlIdentifier name;
+  private final SqlNodeList paramNames;
+  private final SqlNodeList paramValues;
 
-  /** Creates a SqlExecMacro. */
-  public SqlExecMacro(SqlParserPos pos, SqlIdentifier name) {
+  /**
+   * Create an {@code SqlExecMacro}.
+   *
+   * @param pos  Parser position, must not be null
+   * @param name  Name of the macro
+   * @param paramNames  List of parameter names
+   * @param paramValues  List of parameter values
+   */
+  public SqlExecMacro(SqlParserPos pos, SqlIdentifier name,
+      SqlNodeList paramNames, SqlNodeList paramValues) {
     super(pos);
     this.name = Objects.requireNonNull(name);
+    this.paramNames = paramNames;
+    this.paramValues = paramValues;
   }
 
   @Override public SqlOperator getOperator() {
     return OPERATOR;
   }
 
   @Override public List<SqlNode> getOperandList() {
-    return ImmutableNullableList.of(name);
+    // the list of paramNames could be empty
+    return ImmutableNullableList.of(name, paramValues, paramNames);
   }
 
   @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
     writer.keyword("EXECUTE");
     name.unparse(writer, leftPrec, rightPrec);
+    if (paramValues.size() == 0) {
+      return;
+    }
+    SqlWriter.Frame frame = writer.startList("(", ")");
+    if (paramNames.size() == 0) {
+      for (SqlNode e : paramValues) {
+        writer.sep(",", false);
+        e.unparse(writer, 0, 0);
+      }
+    } else {
+      for (int i = 0; i < paramNames.size(); i++) {
+        writer.sep(",", false);
+        paramNames.get(i).unparse(writer, 0, 0);
+        writer.keyword("=");
+        paramValues.get(i).unparse(writer, 0, 0);
+      }
+    }
+    writer.endList(frame);
+
```

Comment:
nit: remove extra newline

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/108/commits/20c4b40c220d3429b0f091cd09306a0e0043d473